sub_peers.js

```diff
@@ -125,7 +197,16 @@ TChannelSubPeers.prototype.chooseLinearPeer = function chooseLinearPeer(req) {
     for (var i = 0; i < hosts.length; i++) {
         var hostPort = hosts[i];
         var peer = self._map[hostPort];
-        if (!req || !req.triedRemoteAddrs || !req.triedRemoteAddrs[hostPort]) {
+
+        var shouldSkip = req && req.triedRemoteAddrs && req.triedRemoteAddrs[hostPort];
+        if (this.hasMinConnections) {
+            var notEnoughPeers = this.currentConnectedPeers < this.minConnections;
```

Comment:
So we'll prefer to initiate a new connection, and wait for its init handshake to complete, before we send the current rpc?  Can't we decouple that so that we don't defer sending the current rpc just to open some new connections?